python/GafferUI/SpreadsheetUI/_Algo.py

```diff
@@ -111,6 +111,28 @@ def addColumn( spreadsheet, plug ) :
 # UI Helpers
 # ----------
 
+## \todo Needs UndoScope removing
+def setPlugValues( plugs, value ) :
+
+	if not plugs :
+		return
+
+	with Gaffer.UndoScope( next( iter( plugs ) ).ancestor( Gaffer.ScriptNode ) ) :
+		for plug in plugs :
+			if not Gaffer.MetadataAlgo.readOnly( plug ) and plug.settable() :
+				plug.setValue( value )
+
+## \todo Needs UndoScope removing
+def deleteRows( rowPlugs ) :
+
+	if not rowPlugs :
+		return
+
+	with Gaffer.UndoScope( next( iter( rowPlugs ) ).ancestor( Gaffer.ScriptNode ) ) :
+		for row in rowPlugs :
+			if not Gaffer.MetadataAlgo.readOnly( row ) :
+				row.parent().removeChild( row )
```

Comment:
It was more about highlighting all the boilerplate that we end up writing in so many classes. We've talked before about being able to make undoable methods on the fly, etc... It felt like at least separating them out made it easier to see the extent of 'pointless functions we keep writing over and over' in one place. If they're embedded in `_PlugTableView` its pretty hard to assess. I bet we have similar stuff in a _lot_ of the UI. Pretty much everything under `UI Helpers` is either: 
 - Wrap in an undo scope
 - Check we're actually allowed to do something

Reply:
Summary of earlier chat (possibly biased) :

- We both agree there is a "missing layer" or "missing algo" somewhere, to reduce boilerplate in UI code.
- I don't think these UI helpers are good examples of what that will look like, so don't think separating them out helps.
- Tom things separating them out helps because it goads me into thinking about what it _should_ like ;)
- We think removing the `ScriptNode` argument from `UndoScope` will make undo boilerplate easier/automatable.
- Generally `isFooAllowed()` should be a separate step performed before `foo()`, rather than a single `doAsMuchFooIsAllowed()` method. This keeps `foo()` useable in non-ui contexts.
- We will open a separate issue to pursue boilerplate improvements.